thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java

```diff
@@ -317,4 +333,124 @@ public Response tuneAlertFilter(@PathParam("function_id") String id,
     }
     return Response.ok(alertFilter.toString()).build();
   }
+
+
+  private List<Map<String, String>> listAllTuningParameters(JSONObject tuningJSON) throws JSONException {
+    List<Map<String, String>> tuningParameters = new ArrayList<>();
+    Iterator<String> jsonFieldIterator = tuningJSON.keys();
+    Map<String, List<String>> fieldToParams = new HashMap<>();
+
+    int numPermutations = 1;
+    while(jsonFieldIterator.hasNext()){
+      String field = jsonFieldIterator.next();
+
+      if(field != null && !field.isEmpty()){
+        // JsonArray to String List
+        List<String> params = new ArrayList<>();
+        JSONArray paramArray = tuningJSON.getJSONArray(field);
+        if(paramArray.length() == 0){
+          continue;
+        }
+        for(int i = 0; i < paramArray.length(); i++){
+          params.add(paramArray.get(i).toString());
+        }
+        numPermutations *= params.size();
+        fieldToParams.put(field, params);
+      }
+    }
+
+    List<String> fieldList = new ArrayList<>(fieldToParams.keySet());
+    for(int i = 0; i < numPermutations; i++){
+      Map<String, String> combination = new HashMap<>();
+      int index = i;
+      for(String field : fieldList) {
+        List<String> params = fieldToParams.get(field);
+        combination.put(field, params.get(index % params.size()));
+        index /= params.size();
```

Comment:
I checked the paramArray.length() in 366-368. I will continue the iteration if it is 0. Should I still check for the params size?